.github/workflows/main.yml

```diff
@@ -141,9 +149,18 @@ jobs:
         with:
           push: true
           tags: ${{ steps.meta.outputs.tags }}
+      - name: Build and Push Preview Docker Image
+        if: steps.get-versions.outputs.has_preview == 'true'
+        uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83
+        with:
+          push: true
+          platforms: ${{ env.PLATFORMS }}
+          build-args: |
+            APOLLO_ROUTER_VERSION=${{ steps.get-versions.outputs.preview_version }}
+          tags: |
+            ${{ env.NAMESPACED_REGISTRY }}:${{ github.event_name == 'pull_request' && format('PR{0}-router-preview', github.event.number) || 'router-preview' }}
```

Comment:
In the same way here, if we use the metadata action, I think that's the only thing that needs to change, so this should simplify things here